File: WavoProjects.UI/WavoProjects/src/app/models/project.ts
import { Team } from "./team";
import { Priority } from "./priority";
import { TeamMember } from "./team-member";

export class Project {
  id: number;
  name: string;
  description?: string;
  teamId?: number;
  team?: Team;
  priorityId?: number;
  priority?: Priority;
  sortOrder?: number;
  projectOwnerId?: number;
  projectOwner?: TeamMember;
  startedOn?: Date;
  createdOn: Date;
  updatedOn: Date;

  constructor(obj: any) {
    Object.keys(obj).forEach(key => {
        this[key] = obj[key];
    });

    if(obj['team'] != null) {
        this.team = new Team(obj['team']);
    }
  }
}

export const ExampleProjects: Project[] = [
    {
        id: 1,
        name: "T-Shirt Cannon Robot",
        priorityId: 1,
        priority: {
            id: 1,
            name: "High"
        },
        teamId: 1,
        team: {
            id: 1,
            name: "Design/CAD",
            color: "#4700d8",
            backgroundColor: "#4700d816"
        },
        sortOrder: 1,
        createdOn: new Date(),
        updatedOn: new Date(),
    },
    {
        id: 2,
        name: "Chairman's Video",
        priorityId: 2,
        priority: {
            id: 2,
            name: "Medium"
        },
        teamId: 2,
        team: {
            id: 2,
            name: "Business",
            color: "#ef0000",
            backgroundColor: "#ef000016"
        },
        sortOrder: 2,
        createdOn: new Date(),
        updatedOn: new Date(),
    },
    {
        id: 3,
        name: "Signage",
        priorityId: 3,
        priority: {
            id: 3,
            name: "Low"
        },
        teamId: 2,
        team: {
            id: 3,
            name: "Media",
            color: "#00ff2d",
            backgroundColor: "#00ff2d"
        },
        sortOrder: 3,
        createdOn: new Date(),
        updatedOn: new Date(),
    }
];

File: WavoProjects.Api/DatabaseModels/Project.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System;
using System.Collections.Generic;
using System.Text;
using WavoProjects.Api.DatabaseModels;

namespace WavoProjects.Api.DatabaseModels
{
    public class Project
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int? TeamId { get; set; }
        public int? PriorityId { get; set; }
        public int? SortOrder { get; set; }
        public DateTime? StartedOn { get; set; }
        public int? ProjectOwnerId { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime? UpdatedOn { get; set; }

        public virtual Team Team { get; set; }
        public virtual Priority Priority { get; set; }
        public virtual TeamMember ProjectOwner { get; set; }
    }

    public partial class ProjectConfiguration : IEntityTypeConfiguration<Project>
    {
        public void Configure(EntityTypeBuilder<Project> entity)
        {
            entity.Property(i => i.Id).ValueGeneratedOnAdd();
            entity.Property(i => i.Name).HasMaxLength(500);
            entity.Property(i => i.Description).HasMaxLength(2000);
            entity.ToTable("Project");


            entity.HasData(
                new Project { Id = 1, Name = "T-Shirt Cannon Robot", Description = "T-Shirt Cannon Robot", TeamId = 3, PriorityId = 1, SortOrder = 1, CreatedOn = DateTime.Now, UpdatedOn = DateTime.Now },
                new Project { Id = 2, Name = "Sponsor Thank Yous", Description = "Sponsor Thank Yous", TeamId = 1, PriorityId = 1, SortOrder = 2, CreatedOn = DateTime.Now, UpdatedOn = DateTime.Now },
                new Project { Id = 3, Name = "Chairman's Presentation", Description = "Chairman's Presentation", TeamId = 2, PriorityId = 1, SortOrder = 3, CreatedOn = DateTime.Now, UpdatedOn = DateTime.Now },
                new Project { Id = 4, Name = "Organize Electrical Toolbox", Description = "Organize Electrical Toolbox", TeamId = 4, PriorityId = 1, SortOrder = 4, CreatedOn = DateTime.Now, UpdatedOn = DateTime.Now },
                new Project { Id = 5, Name = "Make New Router Vacuum Table", Description = "Make New Router Vacuum Table", TeamId = 6, PriorityId = 1, SortOrder = 5, CreatedOn = DateTime.Now, UpdatedOn = DateTime.Now },
                new Project { Id = 6, Name = "Develop Path Planning", Description = "Develop Path Planning", TeamId = 4, PriorityId = 1, SortOrder = 6, CreatedOn = DateTime.Now, UpdatedOn = DateTime.Now }
            );
        }
    }
}
